discovery.md

```diff
@@ -41,19 +41,8 @@ a Discovery Endpoint could choose to be implemented as part of a Service, or
 Event Producer, or it could be acting as an independent aggregrator of this
 metadata. This implementation detail will be transparent to consumers.
 
-There are several discovery use cases to consider from the viewpoint of event
-consumers.
-
-1. What services are available, and what event types do they generate?
-2. What event types are available, and from which services?
-
-The second case becomes relevant if multiple services support the same event
-types. Use case 1 is likely the dominant use case. Given the example of a public
-cloud provider where all services generate events, there might be dozens of
-sources and hundreds of event types. The discovery funnel of services first,
-then event types helps users navigate without having to see large lists of event
-types. Both of these cases show the importance of using filters in the discovery
-API to narrow down the selection of available events.
+The main usecase to condider from the viewpoint of the event consumer is what
+services are available, and what event types do they generate?
```

Comment:
Why do we drop the second use case entirely? I think it is still relevant.

Reply:
I dropped it because the answer to the question is suppose to be calls to the /types endpoint, which I dropped.

Reply:
It does beg the question... how does someone get a list of all of the 'types'?  Or is that even a valid question? Should they be able to ask for the list of all "sourcetemplates" too?  With the rename of "types" to "events", is "type" no longer special? Should it be?

I like the idea of treating all attributes equally. So, perhaps we should ask if getting the list of types is really a query people want to do?  Or is it more likely they people have a type in mind (e.g. com.example.db.created) and are looking for services that emit that?  Is a list of random strings (types) something that's interesting and useful?

Reply:
I guess if someone can explain why the list of types is useful then that would help us define the `/types` API that this PR is removing.